language/move-prover/boogie-backend/src/options.rs

```diff
@@ -153,13 +153,19 @@ impl BoogieOptions {
     }
 
     /// Returns command line to call boogie.
-    pub fn get_boogie_command(&self, boogie_file: &str) -> Vec<String> {
+    pub fn get_boogie_command(&self, boogie_file: &str) -> anyhow::Result<Vec<String>> {
         let mut result = if self.use_exp_boogie {
             // This should have a better ux...
             vec![read_env_var("EXP_BOOGIE_EXE")]
         } else {
             vec![self.boogie_exe.clone()]
         };
+
+        // If we don't have a boogie executable, nothing will work
+        if result.iter().all(|path| path.is_empty()) {
+            return Err(anyhow::Error::msg("No boogie executable set.  Please set BOOGIE_EXE"));
```

Comment:
Nit: converting this to an `anyhow::bail` should work here, and should be cleaner I think.
```suggestion
            anyhow::bail!("No boogie executable set.  Please set BOOGIE_EXE");
```

Reply:
True, was wondering because we've erred away from anyhow and bail in the main diem source.